C/c4CAPI.cc

```diff
@@ -278,19 +284,21 @@ bool c4coll_isValid(C4Collection* coll) noexcept {
 }
 
 C4CollectionSpec c4coll_getSpec(C4Collection *coll) noexcept {
+    // Unlike the others, this continues to return valid data even
+    // after invalidation, so skip the validity check
     return coll->getSpec();
 }
 
 C4Database* c4coll_getDatabase(C4Collection *coll) noexcept {
-    return coll->isValid() ? coll->getDatabase() : nullptr;
+    return _usuallyTrue(c4coll_isValid(coll)) ? coll->getDatabase() : nullptr;
```

Comment:
Won't coll->getDatabase() throw if !coll->isValue(). ?

Reply:
That would be a bad move.  This method is marked noexcept.  Actually in this case it will just return null anyway, but for consistency sake I used this.

Reply:
yes, we want noexcept, but if you catch it as all other c4 functions do, would it not work? It seems to be double protections and defeats the original design of C4Collection

Reply:
There is nothing to actually catch in this case, but I also want to protect against a null `coll` to avoid a SIGSEGV